FeatherCoin/Feathercoin
# commits from bitcoin/master

ui: Compile boost::signals2 only once
https://github.com/bitcoin/bitcoin/pull/13634

tests: Replace usage of tostring() with tobytes()
https://github.com/bitcoin/bitcoin/pull/13963

Remove unused dummy_tx variable from FillPSBT
https://github.com/bitcoin/bitcoin/pull/13962

trivial: Removes unsed `CBloomFilter` constructor.
https://github.com/bitcoin/bitcoin/pull/13948

tests: Fix extended functional tests fail
https://github.com/bitcoin/bitcoin/pull/13965

Fix PSBT deserialization of 0-input transactions
https://github.com/bitcoin/bitcoin/pull/13960

Additional safety checks in PSBT signer
https://github.com/bitcoin/bitcoin/pull/13917

ci: Add Appveyor CI
https://github.com/bitcoin/bitcoin/pull/13964

[trivial] Fix typo in CDiskBlockPos struct's ToString
https://github.com/bitcoin/bitcoin/pull/13974

refactor: use fs:: over boost::filesystem::
https://github.com/bitcoin/bitcoin/pull/13982

[RPC] Remove getinfo deprecation warning
https://github.com/bitcoin/bitcoin/pull/13891

rpc: Add submitheader
https://github.com/bitcoin/bitcoin/pull/13399

[trivial] Fix slightly confusing mispelling in feature_blocksdir.py log message
https://github.com/bitcoin/bitcoin/pull/13985

[build] Add risc-v support to gitian
https://github.com/bitcoin/bitcoin/pull/13665

appveyor: fetch the latest port data
https://github.com/bitcoin/bitcoin/pull/13997

depends: fix qt determinism
https://github.com/bitcoin/bitcoin/pull/14000

[gui] Make proxy icon from statusbar clickable
https://github.com/bitcoin/bitcoin/pull/13248

[qt] TransactionView: highlight replacement tx after fee bump
https://github.com/bitcoin/bitcoin/pull/12818

[wallet] couple of walletcreatefundedpsbt fixes
https://github.com/bitcoin/bitcoin/pull/13968

Use new Qt5 connect syntax
https://github.com/bitcoin/bitcoin/pull/13529

Add const modifier to HTTPRequest methods
https://github.com/bitcoin/bitcoin/pull/14006

Add PSBT documentation
https://github.com/bitcoin/bitcoin/pull/13941

[walletdb] don't report minversion wallet entry as unknown
https://github.com/bitcoin/bitcoin/pull/13967

Add checks for settxfee reasonableness
https://github.com/bitcoin/bitcoin/pull/13988

contrib: add gitian build support for github pull request
https://github.com/bitcoin/bitcoin/pull/13765

p2p: Remove dead code for nVersion=10300
https://github.com/bitcoin/bitcoin/pull/14025

Avoid locking cs_main in some wallet RPC
https://github.com/bitcoin/bitcoin/pull/12559

qa: Add TestNode::assert_debug_log
https://github.com/bitcoin/bitcoin/pull/14024

Add CMerkleTx::IsImmatureCoinBase method
https://github.com/bitcoin/bitcoin/pull/13631

Return the script type from Solver
https://github.com/bitcoin/bitcoin/pull/13429

p2p: Clarify control flow in ProcessMessage
https://github.com/bitcoin/bitcoin/pull/13946

util: Replace boost::signals2 with std::function
https://github.com/bitcoin/bitcoin/pull/13961

tests: Add usage note to check-rpc-mappings.py
https://github.com/bitcoin/bitcoin/pull/13707

Show "bip125-replaceable" flag, when retrieving mempool entries
https://github.com/bitcoin/bitcoin/pull/12676

Add compile time checking for cs_main runtime locking assertions
https://github.com/bitcoin/bitcoin/pull/13083

Move cs_main locking annotations from .cpp to .h
https://github.com/bitcoin/bitcoin/pull/14063

qa: Use assert not BOOST_CHECK_* from multithreaded tests
https://github.com/bitcoin/bitcoin/pull/14069

BIP 158: Compact Block Filters for Light Clients
https://github.com/bitcoin/bitcoin/pull/12254

build: generate MSVC project files via python script
https://github.com/bitcoin/bitcoin/pull/14062

qa: Stop txindex thread before calling destructor
https://github.com/bitcoin/bitcoin/pull/14071

Docs: Fix help message typo optiona -> optional
https://github.com/bitcoin/bitcoin/pull/14056

travis: Run unit tests --with-sanitizers=undefined
https://github.com/bitcoin/bitcoin/pull/14036

Make IS_TRIVIALLY_CONSTRUCTIBLE consistent on GCC < 5, don't patch clang
https://github.com/bitcoin/bitcoin/pull/14031

Remove ambiguity in construction of prevector
https://github.com/bitcoin/bitcoin/pull/14030

test: Add testing of value_ret for SelectCoinsBnB
https://github.com/bitcoin/bitcoin/pull/13861

Report minfeefilter value in getpeerinfo rpc
https://github.com/bitcoin/bitcoin/pull/13987

contrib: Adjust output to current test format
https://github.com/bitcoin/bitcoin/pull/13935

Remove accounts rpcs
https://github.com/bitcoin/bitcoin/pull/14023

Mark single-argument constructors "explicit"
https://github.com/bitcoin/bitcoin/pull/13769

Remove redundant assignments (dead stores)
https://github.com/bitcoin/bitcoin/pull/13767

[Tests] Make combine_logs.py handle multi-line logs
https://github.com/bitcoin/bitcoin/pull/14051

PSBT key path cleanups
https://github.com/bitcoin/bitcoin/pull/13723

qa: Add some actual witness in rpc_rawtransaction
https://github.com/bitcoin/bitcoin/pull/14052

fix walletcreatefundedpsbt deriv paths, add test
https://github.com/bitcoin/bitcoin/pull/14055

tests: Fix accidental trunction from int to bool
https://github.com/bitcoin/bitcoin/pull/14093

utils: drop boost::interprocess::file_lock
https://github.com/bitcoin/bitcoin/pull/13862

Remove the boost/algorithm/string/case_conv.hpp dependency
https://github.com/bitcoin/bitcoin/pull/13671

Explicitly initialize prevector _union
https://github.com/bitcoin/bitcoin/pull/14028

Add tests for RPC help
https://github.com/bitcoin/bitcoin/pull/14020

tx pool: Avoid passing redundant hash into addUnchecked (scripted-diff)
https://github.com/bitcoin/bitcoin/pull/13792

validation: Log FormatStateMessage on ConnectBlock error in ConnectTip
https://github.com/bitcoin/bitcoin/pull/14097

docs: Fix broken Doxygen comments
https://github.com/bitcoin/bitcoin/pull/14103

util: Add Clang thread safety annotations for variables guarded by cs_args
https://github.com/bitcoin/bitcoin/pull/13126

[wallet] Kill accounts
https://github.com/bitcoin/bitcoin/pull/13825

[contrib] Support ARM and RISC-V symbol check
https://github.com/bitcoin/bitcoin/pull/13724

Add README.md to linux release tarballs
https://github.com/bitcoin/bitcoin/pull/14037

qa: Use named args in validation acceptance tests
https://github.com/bitcoin/bitcoin/pull/14101

Don't close old debug log file handle prematurely when trying to re-open (on SIGHUP)
https://github.com/bitcoin/bitcoin/pull/13159

blockfilter: Avoid out-of-bounds script access.
https://github.com/bitcoin/bitcoin/pull/14073

tests: Add missing locking annotations and locks (g_cs_orphans)
https://github.com/bitcoin/bitcoin/pull/14108

tests: Don't assert(...) with side effects
https://github.com/bitcoin/bitcoin/pull/14088

Add AssertLockHeld assertions in CWallet::ListCoins
https://github.com/bitcoin/bitcoin/pull/10605

Make LOCK, LOCK2, TRY_LOCK work with CWaitableCriticalSection
https://github.com/bitcoin/bitcoin/pull/11640

scripted-diff: Small locking rename
https://github.com/bitcoin/bitcoin/pull/11599

Test rpc_help.py failed: Check whether ZMQ is enabled or not.
https://github.com/bitcoin/bitcoin/pull/14122

Notes about control port and read access to cookie
https://github.com/bitcoin/bitcoin/pull/14120
